### DISCUSSION TITLE
Bug - Fonctionnement du délestage

### DISCUSSION THREAD

**Discussion by Balooforever** [jmcollin78/versatile_thermostat#405] [Q&A]
Bonjour,

Tout d'abord, merci pour ce super développement, je suis entrain de basculer toute la partie chauffage qui était sur jeedom depuis des années.

J'ai un seul petit souci : j'ai configurer le délestage avec une valeur de 9000W pour un pole de radiateur de 4500W.

Le souci, c'est qu'au bout du cycle de chauffe, au lieu de continuer à chauffer, le radiateur se mets en sécurité (4500+4500+les quelques Watts > 9000) alors que la vrai consommation dans les 5000. La sécurité saute au bout d'une dizaine de seconde (donc mon switch fait un On/Off dans un interval de 10 secondes..)

Est-il possible de modifier le comportement ?

Merci

**Comment by Balooforever** [comment]
Concernant le délestage, j'ai une interrogation. Un vterm est lancé, si il n'y a pas ré interrogation lors du dépassement de la valeur.

Je m'explique, si on lance 4500W de chauffage pour 5 minutes (cycle vterm classique) mais que je branche ma voiture électrique, nous dépassons la valeur max prévu à ne pas dépasser par le chauffage. Dans ce cas précis, le chauffage ne se coupe et continu à finir son cycle.

Es ce un comportement normal ?

Merci

**Comment by jmcollin78** [comment]
Hello,

> Es ce un comportement normal ?

Oui c'est normal. Ca ne fonctionne pas comme ça. Le fonctionnement est :
1. lorsque VTherm décide d'allumer un radiateur, il vérifie qu'il a la puissance disponible,
2. si il ne l'a pas, il passe en délestage

mais si il est déjà allumé sur son cycle, il n'y a pas moyen de stopper le cycle. Le dépassement sera pris en compte au prochain cycle (au prochaine démarrage après la phase d'arrêt).

C'est pour ça que c'est pas une science exacte et qu'il n'y a pas une garantie absolue qu'à chaque instant ca ne dépassera pas.

L'autre cas, c'est quand la remontée de la puissance se fait toutes les 10 sec et que 2 radiateurs vont démarrer dans ces 10 sec.

**Comment by jmcollin78** [comment]
Je vais transformer cette issue en discussion, ca peut intéresser pas mal de monde.

**Comment by Balooforever** [comment]
Aucun souci, je ne suis pas un grand utilisateur de Github :)

Pour la partie cycle, ce n'est pas très grave, il suffit de prendre une marge suffisante (c'est lié à ma plaque à induction, je voulais éviter le gros appel de courant lorsque je l'allume mais finalement à l'usage elle ne consomme pas tant que ça :)) et je peux surtout couper via une automatisation si besoin .. :)

Je confirme j'ai bien une remontée de puissance toute les 10 secondes (avec un Zlinky). Par contre, c'est dommage que lors de la fin du cycle et du recalcule de la puissance disponible il ne soustrait pas sa puissance qu'il utilise actuellement, cela donne des choses un peu étrange (un du coup une coupure et un allumage de mes radiateurs) : 

![image](https://github.com/jmcollin78/versatile_thermostat/assets/48917118/7fa0998e-86d1-4252-95ac-9b47abcd0d40)

Ne pas trop tenir compte des températures, je suis toujours en cours d'étalonnage :D

**Reply by jmcollin78** [reply]
Y a pas de légende donc je ne sais pas trop ce que je vois sur ton graphe.

**Reply by Balooforever** [reply]
S'était pour illustrer, mon vterm se mets en sécurité à chaque "recalcule" car lui même + lui même entraine un dépassement de la valeur cible.

Pour (essayer) de mieux expliquer : 

Il fait par exemple 15 degrés dans ma pièce.
Le Vterm se mets à chauffer sur la consigne ECO (car je suis absent) de 17 degré. Je rentre à peut prêt au même moment, le chauffage passe en CONFORT. A ce moment là, au lieu de faire ECO -> CONFORT, le vterm va faire ECO -> SECURITE -> CONFORT car il va calculer qu'il ne peut pas lancer son cycle de chauffage car il va dépasser la valeur de délestage, sauf que c'est lui même qui fait que la puissance demandé est haute.

Est-il possible, dans le calcul de faire : si le chauffage est déjà ON, alors soustraire sa valeur en Watt du calcul de délestage ? Si impossible, je vais le faire via automation en modifiant ma valeur de puissance max si le chauffage est ON mais c'est dommage :)

**Reply by jmcollin78** [reply]
Le mode sécurité dans VTherm c'est autre chose. C'est quand le thermomètre n'envoie plus de température. Donc, je suppose que tu parles du délestage et non pas de la sécurité. Le délestage consiste à changer la température cible pour une valeur assez basse, l'idée étant que le radiateur s'éteigne ou limite sa consommation.
Si on en phase avec ça, je continue.

> ECO -> Delestage -> CONFORT car il va calculer qu'il ne peut pas lancer son cycle de chauffage car il va dépasser la valeur de délestage, sauf que c'est lui même qui fait que la puissance demandé est haute sauf que c'est lui même qui fait que la puissance demandé est haute.

C'est pas comme ça. C'est pas la puissance demandée qui est utilisée mais la puissance réelle consommée. Tant que le radiateur n'est pas allumé, il ne peut pas être dans la puissance consommée. Ou je ne comprends pas ce que tu veux dire.
Jamais on ne calcule une puissance demandée, mais on compare la puissance courante + puissance du radiateur qu'on veut allumer par rapport au seuil.

> Est-il possible, dans le calcul de faire : si le chauffage est déjà ON, alors soustraire sa valeur en Watt du calcul de délestage ? Si impossible, je vais le faire via automation en modifiant ma valeur de puissance max si le chauffage est ON mais c'est dommage :)

Si il est ON, on ne va pas tenter de l'activer à nouveau donc on ne fait pas le check qui est au-dessus. Donc ca n'a pas de sens comme question.

J'spère que c'est plus clair.

**Reply by Balooforever** [reply]
Avec des chiffres ce sera peut être plus clair : 

Mon pole de radiateur consomme 4500W
Mon entrée  "limite en Watts" est a 8500W

Le Vterm se lance pour un cycle de chauffe.
Au bout de 5 minutes (durée d'un cycle), avec une consommation instantanée de mon capteur a 6000W, au lieu de continuer à chauffer, il se mets en délestage à 14 dégrée.
Dès qu'il reçoit une nouvelle valeur de ma consommation instantanée (donc sans ses propres 4500W, donc 1500W restant), il se remet à chauffer. C'est ce que j'essayais de te montrer sur ma capture : il est tout seul à chauffer au moment ou il passe a 14 degrés et on est très loin de la limite : il ne devrait pas s'arrêter en passant par le 14 degrés, au pire il devrait juste s'arrêter mais pas délester.

Je me suis donc dit qu'il ne prenait pas en compte sa propre consommation lorsqu'il essayait de lancer un seconde "cycle" après le 1er.

Il me fait la même chose si je passe de confort alors qu'il chauffe au mode boost : il passe par la case délestage "14 degrés" avant de se remettre à chauffer. A l'usage ça ne dérange pas mais le double clic de mon switch est un peu embêtant.

Avec des captures : 

Ma consommation avant allumage : 
![Capture d’écran 2024-03-12 à 17 44 35](https://github.com/jmcollin78/versatile_thermostat/assets/48917118/03750c38-34b2-49ca-80cb-158fcbbedc91)

Allumage du vterm : 
![image](https://github.com/jmcollin78/versatile_thermostat/assets/48917118/7117159d-90d8-4db9-9859-6ff6e316d0d2)
![Capture d’écran 2024-03-12 à 17 45 07](https://github.com/jmcollin78/versatile_thermostat/assets/48917118/5bd02f41-73f0-4bb1-8c43-3cf91660de5d)


Passage en Boost : 
![Capture d’écran 2024-03-12 à 17 45 47](https://github.com/jmcollin78/versatile_thermostat/assets/48917118/51d4da92-e6a3-432c-886a-85681d70115c)

Pourtant la consommation instantané (à cause de ce radiateur) est de : 
![Capture d’écran 2024-03-12 à 17 45 07](https://github.com/jmcollin78/versatile_thermostat/assets/48917118/5bd02f41-73f0-4bb1-8c43-3cf91660de5d)

On est donc inférieur a mes 8500W de limite. A moins que je m'y prenne comme un manche (ce n'est pas exclus :D) j'ai l'impression qu'il y a un bug, il ne devrait pas retester la consommation, non ?

**Reply by jmcollin78** [reply]
Quand il est inactif il ne consomme pas. Forcément. Donc tu ne devrais pas avoir 'inactif' ET conso 5872W à cause de ce radiateur. Si c'est le cas, c'est que ta remontée de conso est fausse. Vérifie qu'il est bien étient (donc 0 conso) lorsqu'il est marqué 'inactif'.

> Je me suis donc dit qu'il ne prenait pas en compte sa propre consommation lorsqu'il essayait de lancer un seconde "cycle" après le 1er.

Ca n'a pas de sens. Si il relance un second cycle c'est qu'il est arrêté. Donc, il ne consomme pas, donc il n'est pas dans la conso courante.

Si la limite est de 8500 W, il n'a aucune raison de délester alors que la conso est de 6000 W lorsqu'il est actif. Si il est actif, c'est qu'il avait la puissance nécessaire pour s'activer (c'est à ce moment là qu'est fait le check) donc il n'a pas délester.
Y a pas de délestage alors qu'il est activé. Ce n'est pas possible.

Je ne sais plus trop comment le dire.

En gros, vérifie que ta remontée de conso est bonne et quand le radiateur est éteint elle baisse bien.

**Reply by Balooforever** [reply]
On ne se comprend pas, je suis désolé d'insister :( .. Regarde mes captures, quand il passe "INACTIF" c'est quand je clique sur le BOOST alors qu'il est déjà en chauffe -> il ne devrait pas passer en INACTIF, il devrait juste passer en BOOST. Ma capture s'était pour illustrer la consommation avant qu'il passe en inactif au lieu de passer en boost (j ai une remonté de consommation toute les 10 secondes).

Il consomme 4500W en confort, je veux un boost je le passe en BOOST. Au lieu de passer directement en BOOST, il passe en délestage alors qu'il ne devrait pas, la consommation étant bien inférieur à la valeur cible.

J'ai refais le test et je confirme il y a bien un souci : 

Je chauffe en confort, 4500W, limite à 8500W, conso instantané à environ 5000W (Chauffage du confort + quelques trucs qui reste). Je clic sur BOOST : je passe en délestage INACTIF au lieu de passer en BOOST. Il semble donc calculer en ne prenant pas en compte sa propre consommation.

2eme test : 
Je chauffe en confort, 4500w, limite à 11000W, consommation instant toujours à environ 5000W. Je clic sur BOOST : le VTERM passe en BOOST.

**Reply by jmcollin78** [reply]
ok, j'ai compris, pas facile à se comprendre, tu as bien fait d'insister. Souvent les gens n'ont juste pas compris comment ça marche donc il a fallut s'accorder sur le vocabulaire (sécurité, délestage, radiateur on/actif, ...) avant d'être sur.

Y a peut être un soucis en effet dans le cas où on change de preset alors que le radiateur est actif. J'ouvre un bug du coup pour regarder.

**Reply by Balooforever** [reply]
Génial, merci. il restera le cas ou le radiateur chauffe, arrive à la fin de son "cycle" de 5 minutes -> dans ce cas précis je pense qu'il relance tout de même le calcul mais c'est peut être une interprétation de ma part :)

**Comment by jmcollin78** [comment]
https://github.com/jmcollin78/versatile_thermostat/issues/407
